examples/dask/start-with-dask.ipynb

```diff
@@ -0,0 +1,128 @@
+{
```

Comment:
<p>I thought this block was difficult to understand. For example, "take this list of pairs of number and run them..." was confusing to me.</p><p> </p><p>What do you think about this proposed rewrite?</p><p> </p><blockquote>The Dask <code>distributed</code> client comes with several methods for managing collections of such futures. The code below shows how to use a few of these.</blockquote><blockquote><br></blockquote><blockquote>* <code>client.map(f, x)</code> = run function <code>f</code> once per item in <code>x</code> . Returns a list of futures.</blockquote><blockquote>* <code>client.gather(futures)</code> = Given a list of futures, pull them back to the client. Returns a list of delayed results (in this example).</blockquote><blockquote>* <code>client.compute(delayed_results)</code> = Given a list of delayed results, materialize the results on the client. Returns a list of futures (<code>sync=True</code>) or actual function results (<code>sync=False</code>)</blockquote><blockquote>* <code>.result()</code> = Wait for a future to complete. Returns its actual value.</blockquote><blockquote><br></blockquote><pre class="ql-syntax" spellcheck="false">
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/saturncloud/examples/pull/116/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
What do you think of this? I took what you put and expanded it out a little, although I am not a Dask expert so I might have messed something up

```
The Dask distributed client comes with several methods for managing collections of such futures. The code below shows how to use a few of these.

* client.map(f, x) = run function f once per item in a list-like object x. Returns a list of futures that can then be evaluated.
* client.gather(futures) = Given a list of futures in a Dask cluster, pull them back to the client machine. Returns a list of delayed results (in this example).
* client.compute(delayed_results) = Given a list of delayed results, brings the results on the client machine. Returns a list of futures (sync=True) or actual function results (sync=False)
* .result() = Wait for a future to complete. Returns its actual value.

All together, the code below will take the list of inputs, create lazy evaluations of the exponents on a Dask cluster, gather them onto the client machine, and get the results.
```

Reply:
I would use `client` instead of `client machine`. It's confusing to say `client machine` because of you use a Dask `LocalCluster` (a really common way to work with Dask), then you're not talking about moving data between *machines* but between worker *processes* and the client *processes*.

Instead of "in a Dask cluster", if you want to add more detail to `client.gather()` it would be more precise to say "Given a list of futures held by Dask workers".

I also realize there was a mistake in my previous comment. `client.compute(sync=True)` returns actual values and `client.compute(sync=False)` returns a list of futures (I had reversed them).

I personally think the phrase "create lazy evaluations of the exponents" is confusing.

Reply:
Okay, here are the additional changes, which I hope should cover it (and I pushed it in the branch too):

```
The Dask distributed client comes with several methods for managing collections of such futures. The code below shows how to use a few of these.

* `client.map(f, x)` - run function `f` once per item in a list-like object `x`. Returns a list of futures that can then be evaluated.
* `client.gather(futures)` - Given a list of futures held by Dask workers, pulls them back to the client. Returns a list of delayed results (in this example).
* `client.compute(delayed_results)` - Given a list of delayed results, brings the results on the client. Returns a list of futures (`sync=False`) or actual function results (`sync=True`)
* `.result()` - Wait for a future to complete. Returns its actual value.

All together, the code below will take the list of inputs, converts them to futures held by Dask workers for apply the exponental function, gathers them onto the client machine, and determines the results.
```

Reply:
This is not grammatically correct:

> converts them to futures held by Dask workers for apply the exponental function

I suggest changing that sentence to something like this:

> All together, the code below takes a list of inputs, and uses Dask to parallelize the work executing `lazy_exponent()` on each input.